### PR TITLE
[Modules] Updated Insight MetricAlertsto new dependencies approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Insight MetricAlerts to new dependencies approach

| Pipeline |
| - |
| [Insights: MetricAlerts](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.metricalerts.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/pipelines/staticValidation/module.tests.ps1 [hunk 1]
```diff
@@ -342,7 +342,13 @@ Describe 'Readme tests' -Tag Readme {
                 $expectedColumnsInOrder += @('Description')
 
                 $readMeCategoryIndex = $readMeContent | Select-String -Pattern "^\*\*$paramCategory parameters\*\*$" | ForEach-Object { $_.LineNumber }
-                $readmeCategoryColumns = ($readMeContent[$readMeCategoryIndex] -split '\|') | ForEach-Object { $_.Trim() } | Where-Object { -not [String]::IsNullOrEmpty($_) }
+
```
Comment (eriqua) [inline utilities/pipelines/staticValidation/module.tests.ps1:345]: Could you elaborate on this change? Is it related to the metric alert testing update?

Comment (MrMCake) [reply utilities/pipelines/staticValidation/module.tests.ps1:345]: Hey @eriqua - I'm not sure what the metric alert test is - but Marius introduced a change to the ReadMe gen in main as part of his workstream and which led to this test failing. @ahmadabdalla just introduced a hotfix and regenerated all ReadMe's (#2132) - and this was essentially my fix. I wasn't sure where the error came from and hence fixed it only for the PRs I worked on in the past week. Will open a dedicated PR now.

Comment (MrMCake) [reply utilities/pipelines/staticValidation/module.tests.ps1:345]: Created PR #2133

Comment (MrMCake) [reply utilities/pipelines/staticValidation/module.tests.ps1:345]: This should then also take care of the conflict

Comment (eriqua) [reply utilities/pipelines/staticValidation/module.tests.ps1:345]: Thanks for opening a dedicated PR. Merge conflicts were unfortunately introduced after its merge and should be solved.

Comment (MrMCake) [reply utilities/pipelines/staticValidation/module.tests.ps1]: done

File: modules/Microsoft.Insights/metricAlerts/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
```
Comment (eriqua) [inline modules/Microsoft.Insights/metricAlerts/.test/common/deploy.test.bicep:44]: Shouldn't I also see the readme in the list of changed files (deployment examples)?

Comment (MrMCake) [reply modules/Microsoft.Insights/metricAlerts/.test/common/deploy.test.bicep:44]: You should - and now you do

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**